curriculum/challenges/english/25-front-end-development/quiz-javascript-problem-solving-and-algorithmic-thinking/66edd4bc0ce399f475ad6f9b.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
 
 ---
 
-Placeholder distractor 3
+Sort the string and count occurrences.
 
 #### --answer--
 
-Placeholder answer
+Iterate through the string, counting consecutive occurrences of each character while building answer.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What approach can be used to reverse the digits of an integer? For example, given the input `1234`, the output should be `4321`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Use bitwise operations to reverse the digits.
 
 ---
 
-Placeholder distractor 2
+Convert the integer to a binary number and reverse it.
 
 ---
 
-Placeholder distractor 3
+Use recursion to reverse digits.
 
 #### --answer--
 
-Placeholder answer
+Use a loop to extract digits from the integer and reconstruct the reversed integer.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Write a function that generates Pascal's Triangle up to `n` rows. For example, for `n = 3`, return `[[1], [1, 1], [1, 2, 1]]`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Generate rows by using a function that outputs prime numbers and maps them to each row.
 
 ---
 
-Placeholder distractor 2
+Use a loop to generate each row by multipling adjacent elements from the previous row.
 
 ---
 
-Placeholder distractor 3
+Use `Math.random()` to generate the rows.
 
 #### --answer--
 
-Placeholder answer
+Use a loop to generate each row by adding adjacent elements from the previous row.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which algorithm is most efficient for sorting an array of size n containing numbers from `1` to `n`?
 
 #### --distractors--
 
-Placeholder distractor 1
+Bubble Sort
 
 ---
 
-Placeholder distractor 2
+Merge Sort
 
 ---
 
-Placeholder distractor 3
+Quick Sort
 
 #### --answer--
 
-Placeholder answer
+Cyclic Sort
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which algorithmic approach is best for solving the problem of finding the minimum number of coins for a given amount when only one type of coin is provided?
 
 #### --distractors--
 
-Placeholder distractor 1
+Dynamic Programming
 
 ---
 
-Placeholder distractor 2
+Divide and Conquer
 
 ---
 
-Placeholder distractor 3
+Backtracking
 
 #### --answer--
 
-Placeholder answer
+Greedy Algorithm
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following problems can be solved efficiently using the two-pointer technique?
 
 #### --distractors--
 
-Placeholder distractor 1
+Sorting an array.
 
 ---
 
-Placeholder distractor 2
+Traversing a tree level by level.
 
 ---
 
-Placeholder distractor 3
+Finding the middle element of a linked list.
 
 #### --answer--
 
-Placeholder answer
+Finding two numbers in a sorted array that sum up to a target value.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Given a Roman numeral string, convert it to its corresponding integer value. For example, `"IX"` should return `9`.
```

Comment:
this is one of the projects present in the curriculum already, I don't know if this maybe should be changed

Reply:
@ilenia-magoni let me know if this question needs to be replaced

Reply:
let's ask @naomi-lgbt if these questions that match labs or workshops can be kept or should be replaced

Reply:
@naomi-lgbt should I replace this question or it's fine here?

Reply:
This should be fine to keep. The quiz comes after the project.